lib/Pipeline/InvalidationEvent.cpp

```diff
@@ -0,0 +1,18 @@
+/// \file InvalidationEvent.cpp
+/// \brief Implementation of invalidation events
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pipeline/InvalidationEvent.h"
+
+using namespace pipeline;
+
+llvm::Error InvalidationEvent::run(Runner &Runner) const {
+  Runner::InvalidationMap Map;
+  getInvalidations(Runner, Map);
+  if (auto Error = Runner.getInvalidations(Map); Error)
```

Comment:
* `getInvalidations` is not a great name. `deduceDerivedInvalidations`?
* This whole method makes me a little suspicious. Why should an invalidation event extend its results?
  In case of multiple invalidation events this would also be more efficient.
  These all seem to be responsibilities of the caller of this method. Can't we have the invalidation event *exclusively* compute the set of invalidate target?

Reply:
> this method is what one would do anyway if it had to manually remove the deduced invalidations

I'm not sure I understand.

Reply:
i mean. this method rappresents the entire action of triggering a invalidation event, figure out the dependencies and drop all  those affected. Somewhere this method must be located, and it must be exactly like this. I can move it to another file, but i don't see much a reason to do that. 

this gives you the syntax
```
SomeInvalidaitonEvent(SomeData).apply(Runner);
```
and everything is done as expected

Reply:
will we really ever have multiple events? we will send a single diff or a user triggered invalidation 99% of the time.

if it becomes a issue, we will add the batching

Reply:
> will we really ever have multiple events

Who knows? Maybe at a certain point we will have a queue of events.

`Runner.getInvalidation` is basically an internal thing of the runner, I don't see why it should be called by someone that should only mind about his own business: computing a set of invalidations.